src/containers/pages/FocusInvestigation/map/active/index.tsx

```diff
@@ -153,319 +143,272 @@ export const defaultMapSingleFIProps: MapSingleFIProps = {
   supersetService: supersetFetch,
 };
 
-/** Map View for Single Active Focus Investigation */
-class SingleActiveFIMap extends React.Component<
-  RouteComponentProps<RouteParams> & MapSingleFIProps,
-  FlexObject
-> {
-  public static defaultProps = defaultMapSingleFIProps;
-  constructor(props: RouteComponentProps<RouteParams> & MapSingleFIProps) {
-    super(props);
+interface StatusBadgeProps {
+  plan: Plan;
+}
+
+const StatusBadge = (props: StatusBadgeProps) => {
+  const { plan } = props;
+  const color =
+    (plan && plan.plan_status === PlanStatus.ACTIVE) ||
+    (plan && plan.plan_status === PlanStatus.DRAFT)
+      ? 'warning'
+      : 'success';
+
+  return (
+    <Badge color={color} pill={true}>
+      {plan && plan.plan_status}
+    </Badge>
+  );
+};
+
+interface MarkCompleteLinkProps {
+  plan: Plan;
+}
+
+const MarkCompleteLink = (props: MarkCompleteLinkProps) => {
```

Comment:
- What are your thoughts regarding moving this to the components directory (or a helper module)?
- Please add a docstring here